modules/bm_sim/include/bm_sim/lookup_structures.h

```diff
@@ -0,0 +1,90 @@
+/* Copyright 2013-present Barefoot Networks, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*
+ * Gordon Bailey (gb@gordonbailey.net)
+ *
+ */
+
+#ifndef BM_SIM_INCLUDE_BM_SIM_LOOKUP_STRUCTURES_H_
+#define BM_SIM_INCLUDE_BM_SIM_LOOKUP_STRUCTURES_H_
+
+#include "match_unit_types.h"
+#include "bytecontainer.h"
+
+namespace bm {
+
+//! This class defines an interface for all data structures used
+//! in Match Units to perform lookups. Custom data strucures can
+//! be created by implementing this interface, and creating a
+//! factory class which will create them.
```

Comment:
An example for the doxygen documentation (maybe just overwriting one of the `create` methods) would be nice.